dataactbroker/handlers/fileHandler.py

```diff
@@ -507,6 +552,9 @@ def checkGeneration(self, submission_id = None, file_type = None):
         """
         if submission_id is None or file_type is None:
             submission_id, file_type = self.getRequestParamsForGenerate()
+        # Check permission to submission
+        self.checkSubmissionById(submission_id, file_type)
```

Comment:
I'm thinking this could be a separate story to make submission level permissions look more like our route-level permissions which do use a decorator.  I'm not going to be able to implement this today as part of this PR, but I'll create a story for it.